### PR TITLE
Add rebalance APIs

### PR DESCRIPTION
**Pull request by KMKoushik**
Fixes ENG-1005

### CODE REVIEW

File: pages/_app.tsx [hunk 1]
```diff
@@ -59,8 +59,8 @@ const InitializePrice = React.memo(function InitializePrice() {
   const { setOsqthRefVolIndex } = useCrabV2Store(s => ({ setOsqthRefVolIndex: s.setOsqthRefVolIndex }), shallow)
 
   const updatePrices = React.useCallback(() => {
-    const p1 = oracle.getTwap(SQUEETH_UNI_POOL, OSQUEETH, WETH, 1, true)
-    const p2 = oracle.getTwap(WETH_USDC_POOL, WETH, USDC, 1, true)
+    const p1 = oracle.getTwap(SQUEETH_UNI_POOL, OSQUEETH, WETH, TWAP_PERIOD, true)
```
Comment (aleone) [inline pages/_app.tsx:62]: imo these should prb be spot prices not twaps

File: tests/unit/calmBull.test.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+  /**
+   * Initial values
+   * ETH price = 1000
+   * crabPrice = 500
+   * sqth price = .1
+   * CR = 2
+   * delta = 1
+   */
+  test('Rebalance when ETH price go down', async () => {
```
Comment (aleone) [inline tests/unit/calmBull.test.ts:46]: Leverage rebalance should only be used when crab changes value relative to debt, so think the eth price change doesn't matter as much for this test - the fact we leverage rebalance is really that crab reduces in value relative to debt, so probably best to make that the explicit name of what test is doing

File: tests/unit/calmBull.test.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        },
+      )
+  }
+
+  /**
+   * Initial values
+   * ETH price = 1000
+   * crabPrice = 500
+   * sqth price = .1
+   * CR = 2
```
Comment (aleone) [inline tests/unit/calmBull.test.ts:43]: maybe good to assert that these are true to ensure test is testing what you think

ie assert that cr = 2 before test and that it goes down when eth goes down, and delta is 1 before, goes up when crab + eth changes, etc

File: tests/unit/calmBull.test.ts (hunk 2 above)
Comment (aleone) [reply tests/unit/calmBull.test.ts:46]: basically delta being different from 1 can be fixed by full or leverage rebalances, as long as CR is respected

delta only changes when the crab vs debt value changes, but becomes larger or smaller depending on eth price changes as well (but only changes if there is a change in crab value, which in practice will be 100% of the time :) )

CR changes when eth price changes (which can also have an impact on the crab value to be clear as ETH price changes oSQTH price assuming vol constant)

**Review by aleone — APPROVED** [review]
generally lgtm at first review, few comments to improve tests

File: tests/unit/calmBull.test.ts (hunk 3 above)
Comment (KMKoushik) [reply tests/unit/calmBull.test.ts:43]: Do we have to do that in FE tests? The point of the FE test is to check if the correct rebalance is selected for a given crab price!

**Pull request merged**